lib/_tls_wrap.js

```diff
@@ -253,23 +251,53 @@ function TLSSocket(socket, options) {
 util.inherits(TLSSocket, net.Socket);
 exports.TLSSocket = TLSSocket;
 
-TLSSocket.prototype._init = function(socket) {
-  assert(this._handle);
+var proxiedMethods = [
+  'close', 'ref', 'unref', 'open', 'bind', 'listen', 'connect', 'bind6',
+  'connect6', 'getsockname', 'getpeername', 'setNoDelay', 'setKeepAlive',
+  'setSimultaneousAccepts', 'setBlocking',
 
-  // lib/net.js expect this value to be non-zero if write hasn't been flushed
-  // immediately
-  // TODO(indutny): rewise this solution, it might be 1 before handshake and
-  // represent real writeQueueSize during regular writes.
-  this._handle.writeQueueSize = 1;
+  // PipeWrap
+  'setPendingInstances'
+];
+
+TLSSocket.prototype._wrapHandle = function(handle) {
+  var res;
 
-  var self = this;
   var options = this._tlsOptions;
+  if (!handle) {
+    handle = options.pipe ? createPipe() : createTCP();
+    handle.owner = this;
+  }
 
   // Wrap socket's handle
   var context = options.secureContext ||
                 options.credentials ||
                 tls.createSecureContext();
-  this.ssl = tls_wrap.wrap(this._handle, context.context, options.isServer);
+  res = tls_wrap.wrap(handle, context.context, options.isServer);
+  res._parent = handle;
+  res._reading = handle._reading;
+
+  // Proxy HandleWrap, PipeWrap and TCPWrap methods
+  proxiedMethods.forEach(function(name) {
+    res[name] = function methodProxy() {
+      return handle[name].apply(handle, arguments);
+    };
+  });
```

Comment:
This usually applies to `close` and `connect` methods. Nothing performance critical.